bloom-instance/network/locals.tf

```diff
@@ -0,0 +1,16 @@
+
+//data "aws_default_tags" "current" {}
+
+locals {
+  #tags = var.tags != null ? var.tags : data.aws_default_tags.current.tags
+  default_name = "${var.name_prefix}:${var.name}"
```

Comment:
I try to avoid adding `:` in names. It's easier to query by prefix if we add an additional tag.  Is name_prefix like an environment?

Reply:
The name_prefix is required for namespacing, with an added benefit of making it easy to identify resources at a glance.  Many, if not most, resource names are either globally unique or unique to an account or account/region.  We have multiple instances of resources running in multiple environments within the same account and region, so namespacing with a prefix prevents collisions.  We can't just name an ALB "Public-ALB", for example, because that will prevent us from having more than one version of the templates running at the same time (ie dev, staging, prod).  We also don't want to just name it "${terraform.workspace}-Public-ALB", because if we ever have multiple tenants in the same account (Bloom/Doorway or otherwise), we risk another conflict.  Scoping it to a user-defined prefix combined with the current workspace enables us to generate deterministic but sufficiently varied resource names that virtually eliminate any chance of a conflict now or in the future.

Using colons for separator chars is mainly to provide distinction between other segments that may themselves have hyphens in them already.  It's easier (for me anyway) to grok "doorway-dev:ecs-service:backend" than "doorway-dev-ecs-service-backend", for example.  Not all services support colons in names, however, so I've been replacing them here and there.  I'm not dogmatic about it in any way, so I'm happy to switch over to hyphen or underscore instead.